cmd/forkdetector/internal/storage.go

```diff
@@ -0,0 +1,547 @@
+package internal
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"github.com/wavesplatform/gowaves/pkg/crypto"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+	"go.uber.org/zap"
+	"net"
+)
+
+const (
+	blocksPrefix        byte = iota // Keys to store blocks by its signatures
+	blocksCounterPrefix             // Keys to store the instance of total number of blocks
+	blockNumbersPrefix              // Keys to store the number of block by its signature
+	blockLinksPrefix                // Keys to store the block link by the block number
+	heightsPrefix                   // Keys to store the signatures of blocks by its heights
+	peerLeashPrefix                 // Keys to store the numbers of blocks last seen from peers
+	peerNodePrefix                  // Keys to store peers by its IPs
+)
+
+var (
+	blocksCounterKey = []byte{blocksCounterPrefix}
+	zeroSignature    = crypto.Signature{}
+	maxSignature     = crypto.Signature{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}
```

Comment:
`maxSignature = crypto.Signature{bytes.Repeat([]byte{0xff}, crypto.SignatureSize)}` looks nicer.

Reply:
Won't work. Here we have to initialise array, not slice. Using repeat we have to convert generated slice of bytes into array, which in turn could return an error.